ql/src/java/org/apache/hadoop/hive/ql/plan/PlanUtils.java

```diff
@@ -1244,16 +1247,31 @@ static Map getPropertiesExplain(Properties properties) {
         }
         clone.remove(StatsSetupConst.NUM_ERASURE_CODED_FILES);
       }
-      if (properties.containsKey(TABLE_IS_CTAS)) {
-        if (clone == null) {
-          clone = new HashMap<>(properties);
+      for (String key : FILTER_OUT_FROM_EXPLAIN) {
+        if (properties.containsKey(key)) {
+          if (clone == null) {
+            clone = new HashMap<>(properties);
+          }
+          clone.remove(key);
         }
-        clone.remove(TABLE_IS_CTAS);
       }
       if (clone != null) {
         return clone;
       }
     }
     return properties;
   }
+
+  public static Map<String, String> getPropertiesExplain(Map<String, String> properties, String... propertiesToRemove) {
```

Comment:
i think a better name would be `getPropertiesForExplain`, however, there was already non overloaded version with same name

Reply:
Rename both of them.